envi/archs/amd64/disasm.py

```diff
@@ -376,27 +464,33 @@ def disasm(self, bytez, offset, va):
 
             else:
                 # print("ADDRTYPE", hex(addrmeth))
-                ameth = self._dis_amethods[addrmeth >> 16]
+                ameth = self._dis_amethods[(addrmeth >> 16) & 0x7F]
+                vex_skip = addrmeth & opcode86.ADDRMETH_VEXSKIP
                 # print("AMETH", ameth)
+                if not isvex and vex_skip:
+                    continue
+
                 if ameth is None:
                     raise Exception("Implement Addressing Method 0x%.8x" % addrmeth)
 
                 # NOTE: Depending on your addrmethod you may get beginning of operands, or offset
                 try:
-                    if addrmeth == opcode86.ADDRMETH_I or addrmeth == opcode86.ADDRMETH_J:
+                    if addrmeth in IMM_REQOFFS:
                         osize, oper = ameth(bytez, offset+operoffset, tsize, prefixes, operflags)
 
                         # If we are a sign extended immediate and not the same as the other operand,
                         # do the sign extension during disassembly so nothing else has to worry about it..
                         if len(operands) and tsize != operands[-1].tsize:
                             # Check if we are an explicitly signed operand *or* REX.W
-                            if operflags & opcode86.OP_SIGNED or prefixes & PREFIX_REX_W:
+                            if operflags & opcode86.OP_SIGNED and prefixes & PREFIX_REX_W:
```

Comment:
actual bugfix here?  that seems pretty epic for these addrmeths.  good catch!

Reply:
Oh yea. I noticed this was discrepancy between the comment, amd64, and i386, and caused some decoding issues with things like pextrd_q and shorter forms of add. It's a small fix, but helps ton with sizing.